nemo/collections/asr/parts/submodules/rnnt_loop_labels_computer.py

```diff
@@ -207,20 +230,35 @@ def __init__(
         self._init_confidence_method(confidence_method_cfg=confidence_method_cfg)
         assert self._SOS == self._blank_index  # "blank as pad" algorithm only
 
-        self.use_cuda_graphs = allow_cuda_graphs
-
-        if self.use_cuda_graphs and self.max_symbols is None:
-            logging.warning("Max symbols is None, which is not allowed with Cuda graphs.")
-            self.use_cuda_graphs = False
+        if not allow_cuda_graphs:
+            self.cuda_graphs_mode = None
+        else:
+            # cuda graphs are allowed
+            # check basic requirements for cuda graphs
+            if self.max_symbols is None:
+                logging.warning("Max symbols is None, which is not allowed with Cuda graphs.")
```

Comment:
People don't check warnings often enough. I recommend you throw an exception here.

Reply:
Max symbols is none on some older models if i recall, that means out of the box they will crash to infer due to error for unrelated reason of cuda graphs not supporting None. It should remain a warning, but i agree that instead of crashing, set it to large default value for cuda graphs..

Ie warn that its not supported so instead a default of 10 timesteps or higher is being used for cuda graphs optimization

Reply:
I agree with Som's suggestion, actually. Using a large value like 10 when someone passes in None seems like the right move.

Reply:
Fully agree, thanks! I fixed the behavior.